src/vcf/fixer.cpp

```diff
@@ -210,10 +243,139 @@ namespace ebi
             std::cerr << "DEBUG: line " << line_number << ": fixing duplicate: removing variant: "
             << std::string{line->begin(), line->end()} << std::endl;
         }
+
+        size_t Fixer::remove_duplicate_strings(const std::string &column,
+                                               const std::string &separator)
+        {
+            std::set<std::string> already_present;
+            auto is_value_duplicated = [&](const std::string &value, size_t index) -> bool {
+                bool first_appearance = already_present.insert(value).second;
+                return not first_appearance;
+            };
+            return remove_fields(column, separator, is_value_duplicated);
+        }
+
+        size_t Fixer::remove_duplicate_key_value_pairs(const std::string &column,
+                                                       const std::string &separator,
+                                                       const std::string &key_value_separator,
+                                                       const std::string &empty_value)
+        {
```

Comment:
I have found a way to improve performance and clarity slightly:

1. `map<string, string> values` stores the first occurrence of a key, so it could be renamed to `first_key_occurrence` for clarity.
2. Turn `set<string> fields_to_remove` into a `vector<bool>` whose values will be true if a field shall be removed.
2.1. The `iterator` variable should be renamed to something more descriptive.
3. Use a numerical index to serially loop through `ordered_keys` and `fields_to_remove`, instead of having to search the whole set every time.